cohorts/cohortsRouter.js

```diff
@@ -0,0 +1,96 @@
+const express = require('express');
+const db = require('../config/dbConfig');
+const { ensureValidCohort } = require('../common/middleware');
+
+const router = express.Router();
+
+/** ** POST ROUTE *** */
+router.post('/', (req, res) => {
+  console.log(req.body.length);
+
+  if (!req.body.name || Object.keys(req.body).length !== 1) {
+    res.status(400).json({ error: 'the POST should only contain an object with a name' });
+  } else {
+    db('cohorts')
+      .insert(req.body)
+      .then((count) => {
+        if (!count) {
+          res.status(500).json({ error: 'there was an error adding the cohort' });
+        } else {
+          res.status(201).json({ message: 'the cohort was created' });
+        }
+      })
+      .catch((err) => {
+        res.status(500).json({ error: `there was an error adding the cohort: ${err}` });
+      });
+  }
+});
+
+/** ** GET ROUTES *** */
+router.get('/', (req, res) => {
+  db('cohorts')
+    .then((list) => {
+      if (list.length) {
+        res.status(200).json({ list });
+      } else {
+        res.status(400).json({ message: 'there are no cohorts or there was an error retrieving them' });
+      }
+    })
+    .catch((err) => {
+      res.status(500).json({ error: `there was an error retrieving the cohorts: ${err}` });
+    });
+});
+
+router.get('/:id', ensureValidCohort, (req, res) => {
```

Comment:
A lot of people have stopped using custom middleware (other than error handling but it's not used in the same way). It's really good that you're still getting your reps in.